### PR TITLE
New campaign condition: Contacts campaign

### PR DESCRIPTION
**Pull request by kuzmany**
[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | y
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/4137 https://github.com/mautic/mautic/issues/5246
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
PR added new campaign condition based on campaign membership and date added to campaign.

#### Use cases
- My campaign running some time and I want to add some stuff just for contacts added after some day 
- My campaign running some time and I want to add some stuff  just with contacts added years ago ...
- I want to run campaign just for contacts they aren't in another campaigns

image

#### Steps to test this PR:
1.  Create campagin with new condition Contacts campaign
2.  Test membership and date added limit also

### CODE REVIEW

**Comment by kuzmany** [comment]
Label: Ready to test

File: app/bundles/CampaignBundle/Entity/LeadRepository.php [hunk 1]
```diff
@@ -155,4 +155,38 @@ public function updateLead($fromLeadId, $toLeadId)
...
+                $q->expr()->andX(
+                    $q->expr()->eq('l.lead_id', ':leadId'),
+                    $q->expr()->in('l.campaign_id', $options['campaigns'])
+                )
+            );
+
+        if (!empty($options['dataAddedLimit'])) {
+            $q->andWhere(
+                    $q->expr()->{$options['expr']}('l.date_added', ':dateAdded')
+                )->setParameter('dateAdded', $options['dateAdded']);
```
Comment (escopecz) [inline app/bundles/CampaignBundle/Entity/LeadRepository.php]: Move the 2 lines above 4 spaces to the left.

File: app/bundles/LeadBundle/Config/config.php [hunk 2]
```diff
@@ -596,6 +597,11 @@
                 'class' => 'Mautic\LeadBundle\Form\Type\CampaignEventLeadSegmentsType',
                 'alias' => 'campaignevent_lead_segments',
             ],
+            'mautic.form.type.campaignevent_lead_campaigns' => [
+                'class'     => 'Mautic\LeadBundle\Form\Type\CampaignEventLeadCampaignsType',
```
Comment (escopecz) [inline app/bundles/LeadBundle/Config/config.php]: Use ::class instead of string

File: app/bundles/CampaignBundle/Entity/LeadRepository.php [hunk 3]
```diff
@@ -155,4 +155,38 @@ public function updateLead($fromLeadId, $toLeadId)
...
+        }
+        $q = $this->_em->getConnection()->createQueryBuilder();
+        $q->select('l.campaign_id')
+            ->from(MAUTIC_TABLE_PREFIX.'campaign_leads', 'l');
+        $q->where(
+                $q->expr()->andX(
+                    $q->expr()->eq('l.lead_id', ':leadId'),
+                    $q->expr()->in('l.campaign_id', $options['campaigns'])
+                )
+            );
```
Comment (escopecz) [inline app/bundles/CampaignBundle/Entity/LeadRepository.php]: Move the 5 lines above 4 spaces to the left.

Use `$q->expr()->in('l.campaign_id', ':campaignIds')` and `$q->setParameter('campaignIds', $options['campaigns'], \Doctrine\DBAL\Connection::PARAM_INT_ARRAY)` to make this secure.

Comment (kuzmany) [reply app/bundles/CampaignBundle/Entity/LeadRepository.php]: Let's review now.

**Comment by Maxell92** [comment]
Hello @kuzmany,

could you please fix CS fixer issues?

**Comment by mleffler** [comment]
@Maxell92 Since kuzmany is on vacation, can we do the code formatting for him?  Looks straightforward, and it's unfortunate timing we just realized this when he is out.

**Comment by tony-brandner** [comment]
What happened with this merge? Does this mean it was released?

**Comment by escopecz** [comment]
@tony-brandner you can see the release in the right hand column as the milestone. It was merged to release 2.13.0

**Pull request merged**